### PR TITLE
Onboard repo on async publishing

### PR DESCRIPTION
**Pull request by JohnTortugo**
Relates to: https://github.com/dotnet/arcade/issues/2442

Goal: mitigate `lock on the feed problem` and add further validations. [More details here.](https://github.com/dotnet/arcade/blob/master/Documentation/CorePackages/AsyncPublishing_HowToUse.md)

Test build was here: https://dnceng.visualstudio.com/internal/_build/results?buildId=146900
Test release: https://dnceng.visualstudio.com/internal/_releaseProgress?_a=release-pipeline-progress&releaseId=4312

### CODE REVIEW

**Comment by jcagme** [comment]
@singhsarab could you PTAL?

File: azure-pipelines.yml [hunk 1]
```diff
@@ -1,6 +1,10 @@
 variables:
 - name: _TeamName
   value: UnitTesting
+- name: _PublishUsingPipelines
+  value: true
+- name: _DotNetArtifactsCategory
```
Comment (singhsarab) [inline azure-pipelines.yml]: Not all artifacts built here are for .Net Core.
Just want to access the impact here. There are some artifacts, like the vsix and vsman which get inserted into the devdiv directly.

**Comment by jcagme** [comment]
Yeah this is just the name of a category that we then map to a package feed. That said, if you are doing VS insertion we should know about it. Do you do this on every build? Does this happen on the "Build and Publish" task? I couldn't tell by just eyeballing one of the builds.

File: azure-pipelines.yml (hunk 1 above)
Comment (JohnTortugo) [reply azure-pipelines.yml]: I'm checking this with @singhsarab offline

**Pull request merged**